server/recovery/handler.go

```diff
@@ -0,0 +1,57 @@
+package recovery
+
+import (
+	"fmt"
+	"runtime/debug"
+)
+
+// Wrap wraps a callback with a handler that recovers from a panic, logging and tracking metrics.
+func Wrap(name string, logError func(msg string, keyValuePairs ...any), metrics Metrics, callback func()) func() {
```

Comment:
I think it should be the functions responsibility to deal with panics and not the caller. Do I just wrap every function call in this?

Reply:
It's definitely the functions responsibility and not the caller, but as we have to replicate the same pattern in half a dozen places, this wrapper just makes that easier. I still think it's the function doing the work here: just a slightly fatter function being given to the caller.

Reply:
It's not the function doing the work because I as the caller have to remember to wrap it. If we know the function has to handle panics we should handle them within that function.
I would rather see the recovery code in the function with maybe some explanation of why it exists rather than having every call wrapped. 
What happens when someone comes along and calls the function we are wrapping in another place and does not wrap it?